lib/util/cache/repository/impl/cache-factory.ts

```diff
@@ -0,0 +1,31 @@
+import type { RepositoryCacheType } from '../../../../config/types';
+import { logger } from '../../../../logger';
+import { parseUrl } from '../../../url';
+import type { RepoCache } from '../types';
+import { LocalRepositoryCache } from './local';
+
+export class CacheFactory {
+  private static client: RepoCache | null;
+
+  static get(
+    repository: string,
+    cacheType: RepositoryCacheType = 'local'
+  ): RepoCache {
+    const type = parseUrl(cacheType)?.protocol ?? 'local';
+
```

Comment:
```suggestion
    const type = cacheType.split('://')[0].trim().toLowerCase();
```
couldn't choose between these two

Reply:
removed for now